core/src/test/scala/unit/kafka/utils/JaasTestUtils.scala

```diff
@@ -25,17 +25,35 @@ object JaasTestUtils {
                              keyTab: String,
                              principal: String,
                              debug: Boolean,
-                             serviceName: Option[String]) {
+                             serviceName: String) {
     def toJaasModule: JaasModule = {
+      var module = ""
+      var entries = Map(
+        "principal" -> principal
+      )
```

Comment:
You could perhaps avoid the vars by creating `JaasModule` with the appropriate options within the if-else blocks. The code may be more readable with a set of key-value pairs in each section, keeping the current non-IBM code as-is and adding a similar block for the IBM module.